From: Anke Boersma <demm@kaosx.us>
Date: Sun, 28 Jun 2015 15:46:06 +0000 (+0200)
Subject: Fix compilation with Qt 5.5
X-Git-Url: http://git.subsurface-divelog.org/?p=marble.git;a=commitdiff_plain;h=11c7e96e96c11ae24a485ce34f95925365acc4f4

Fix compilation with Qt 5.5

(cherry picked from commit 92748d6ad115879b38fc3034a0c244b594ca3ba9)
---

diff --git a/src/lib/marble/DiscCache.cpp b/src/lib/marble/DiscCache.cpp
index d6199e6..c7bcc0d 100644
--- a/src/lib/marble/DiscCache.cpp
+++ b/src/lib/marble/DiscCache.cpp
@@ -16,6 +16,7 @@
 #include <QtGlobal>
 #include <QFile>
 #include <QDirIterator>
+#include <QDataStream>
 
 using namespace Marble;
 
diff --git a/src/lib/marble/geodata/data/GeoDataCoordinates.cpp b/src/lib/marble/geodata/data/GeoDataCoordinates.cpp
index 94cda07..d1616fe 100644
--- a/src/lib/marble/geodata/data/GeoDataCoordinates.cpp
+++ b/src/lib/marble/geodata/data/GeoDataCoordinates.cpp
@@ -23,6 +23,7 @@
 #include <QStringList>
 #include <QCoreApplication>
 #include <QAtomicInt>
+#include <QDataStream>
 
 #include "MarbleGlobal.h"
 #include "MarbleDebug.h"
diff --git a/src/lib/marble/geodata/data/GeoDataGeometry.cpp b/src/lib/marble/geodata/data/GeoDataGeometry.cpp
index d0206ee..527ba4f 100644
--- a/src/lib/marble/geodata/data/GeoDataGeometry.cpp
+++ b/src/lib/marble/geodata/data/GeoDataGeometry.cpp
@@ -21,6 +21,8 @@
 
 #include "MarbleDebug.h"
 
+#include <QDataStream>
+
 
 namespace Marble
 {
diff --git a/src/lib/marble/geodata/data/GeoDataLabelStyle.cpp b/src/lib/marble/geodata/data/GeoDataLabelStyle.cpp
index 1ec0c78..eb96091 100644
--- a/src/lib/marble/geodata/data/GeoDataLabelStyle.cpp
+++ b/src/lib/marble/geodata/data/GeoDataLabelStyle.cpp
@@ -12,6 +12,7 @@
 #include "GeoDataLabelStyle.h"
 
 #include <QFont>
+#include <QDataStream>
 
 #include "GeoDataTypes.h"
 
diff --git a/src/lib/marble/geodata/data/GeoDataLatLonAltBox.cpp b/src/lib/marble/geodata/data/GeoDataLatLonAltBox.cpp
index f0bbe6b..54a25bc 100644
--- a/src/lib/marble/geodata/data/GeoDataLatLonAltBox.cpp
+++ b/src/lib/marble/geodata/data/GeoDataLatLonAltBox.cpp
@@ -18,6 +18,8 @@
 
 #include "GeoDataTypes.h"
 
+#include <QDataStream>
+
 namespace Marble
 {
 
diff --git a/src/lib/marble/geodata/data/GeoDataLatLonBox.cpp b/src/lib/marble/geodata/data/GeoDataLatLonBox.cpp
index 5dd7834..ac2c635 100644
--- a/src/lib/marble/geodata/data/GeoDataLatLonBox.cpp
+++ b/src/lib/marble/geodata/data/GeoDataLatLonBox.cpp
@@ -18,6 +18,8 @@
 
 #include "GeoDataTypes.h"
 
+#include <QDataStream>
+
 namespace Marble
 {
 
diff --git a/src/lib/marble/geodata/data/GeoDataLineString.cpp b/src/lib/marble/geodata/data/GeoDataLineString.cpp
index 8864518..e210f60 100644
--- a/src/lib/marble/geodata/data/GeoDataLineString.cpp
+++ b/src/lib/marble/geodata/data/GeoDataLineString.cpp
@@ -18,6 +18,8 @@
 #include "Quaternion.h"
 #include "MarbleDebug.h"
 
+#include <QDataStream>
+
 
 namespace Marble
 {
diff --git a/src/lib/marble/geodata/data/GeoDataLineStyle.cpp b/src/lib/marble/geodata/data/GeoDataLineStyle.cpp
index 2e54493..e2817d2 100644
--- a/src/lib/marble/geodata/data/GeoDataLineStyle.cpp
+++ b/src/lib/marble/geodata/data/GeoDataLineStyle.cpp
@@ -13,6 +13,8 @@
 
 #include "GeoDataTypes.h"
 
+#include <QDataStream>
+
 namespace Marble
 {
 
diff --git a/src/lib/marble/geodata/data/GeoDataMultiGeometry.cpp b/src/lib/marble/geodata/data/GeoDataMultiGeometry.cpp
index 15e8ebe..8dcdfcf 100644
--- a/src/lib/marble/geodata/data/GeoDataMultiGeometry.cpp
+++ b/src/lib/marble/geodata/data/GeoDataMultiGeometry.cpp
@@ -19,6 +19,8 @@
 
 #include "MarbleDebug.h"
 
+#include <QDataStream>
+
 
 namespace Marble
 {
diff --git a/src/lib/marble/geodata/data/GeoDataMultiTrack.cpp b/src/lib/marble/geodata/data/GeoDataMultiTrack.cpp
index 57f35e5..667cefd 100644
--- a/src/lib/marble/geodata/data/GeoDataMultiTrack.cpp
+++ b/src/lib/marble/geodata/data/GeoDataMultiTrack.cpp
@@ -18,6 +18,8 @@
 
 #include "MarbleDebug.h"
 
+#include <QDataStream>
+
 
 namespace Marble
 {
diff --git a/src/lib/marble/geodata/data/GeoDataPolyStyle.cpp b/src/lib/marble/geodata/data/GeoDataPolyStyle.cpp
index 4c5df09..e9fc307 100644
--- a/src/lib/marble/geodata/data/GeoDataPolyStyle.cpp
+++ b/src/lib/marble/geodata/data/GeoDataPolyStyle.cpp
@@ -13,6 +13,8 @@
 
 #include "GeoDataTypes.h"
 
+#include <QDataStream>
+
 namespace Marble
 {
 
diff --git a/src/lib/marble/geodata/data/GeoDataPolygon.cpp b/src/lib/marble/geodata/data/GeoDataPolygon.cpp
index 9ae7091..b20cb3c 100644
--- a/src/lib/marble/geodata/data/GeoDataPolygon.cpp
+++ b/src/lib/marble/geodata/data/GeoDataPolygon.cpp
@@ -15,6 +15,8 @@
 
 #include "MarbleDebug.h"
 
+#include <QDataStream>
+
 
 namespace Marble
 {
diff --git a/src/lib/marble/routing/RoutingModel.h b/src/lib/marble/routing/RoutingModel.h
index 1e00fba..68be229 100644
--- a/src/lib/marble/routing/RoutingModel.h
+++ b/src/lib/marble/routing/RoutingModel.h
@@ -14,7 +14,7 @@
 #include "marble_export.h" 
 #include "GeoDataLineString.h"
 #include "Route.h"
-
+#include <QIODevice>
 #include <QAbstractListModel>
 #include <QTime>
 
diff --git a/src/plugins/render/satellites/SatellitesConfigAbstractItem.h b/src/plugins/render/satellites/SatellitesConfigAbstractItem.h
index 099f619..9729b8e 100644
--- a/src/plugins/render/satellites/SatellitesConfigAbstractItem.h
+++ b/src/plugins/render/satellites/SatellitesConfigAbstractItem.h
@@ -13,6 +13,8 @@
 
 #include <QVector>
 #include <QStringList>
+#include <QHash>
+#include <QVariant>
 
 namespace Marble
 {
diff --git a/src/plugins/runner/cache/CacheRunner.cpp b/src/plugins/runner/cache/CacheRunner.cpp
index ffcacba..15053c1 100644
--- a/src/plugins/runner/cache/CacheRunner.cpp
+++ b/src/plugins/runner/cache/CacheRunner.cpp
@@ -14,6 +14,7 @@
 #include "GeoDataPlacemark.h"
 
 #include <QFile>
+#include <QDataStream>
 
 namespace Marble
 {
diff --git a/src/plugins/runner/json/JsonParser.h b/src/plugins/runner/json/JsonParser.h
index b66b39e..733e030 100644
--- a/src/plugins/runner/json/JsonParser.h
+++ b/src/plugins/runner/json/JsonParser.h
@@ -16,6 +16,7 @@
 #include <QScriptEngine>
 #include <QScriptValue>
 #include <QScriptValueIterator>
+#include <QIODevice>
 
 namespace Marble {
 
diff --git a/src/plugins/runner/pn2/Pn2Runner.h b/src/plugins/runner/pn2/Pn2Runner.h
index aff4b31..6949fe2 100644
--- a/src/plugins/runner/pn2/Pn2Runner.h
+++ b/src/plugins/runner/pn2/Pn2Runner.h
@@ -12,6 +12,8 @@
 
 #include "ParsingRunner.h"
 
+#include <QDataStream>
+
 namespace Marble
 {
 
diff --git a/src/plugins/runner/pnt/PntRunner.cpp b/src/plugins/runner/pnt/PntRunner.cpp
index acbccf3..79b7e78 100644
--- a/src/plugins/runner/pnt/PntRunner.cpp
+++ b/src/plugins/runner/pnt/PntRunner.cpp
@@ -19,6 +19,7 @@
 #include <qmath.h>
 #include <QFile>
 #include <QFileInfo>
+#include <QDataStream>
 
 namespace Marble
 {
diff --git a/tools/dateline/dateline.cpp b/tools/dateline/dateline.cpp
index 9de40a4..97bcd00 100644
--- a/tools/dateline/dateline.cpp
+++ b/tools/dateline/dateline.cpp
@@ -17,6 +17,7 @@
 #include <QFile>
 #include <QVector>
 #include <QStringList>
+#include <QDataStream>
 #include "Quaternion.h"
 
 using namespace Marble;
diff --git a/tools/kml2cache/kml2cache.cpp b/tools/kml2cache/kml2cache.cpp
index e896a50..ad3d405 100644
--- a/tools/kml2cache/kml2cache.cpp
+++ b/tools/kml2cache/kml2cache.cpp
@@ -24,6 +24,7 @@
 #include <QDebug>
 #include <QFile>
 #include <iostream>
+#include <QDataStream>
 
 using namespace std;
 using namespace Marble;
diff --git a/tools/kml2kml/kml2kml.cpp b/tools/kml2kml/kml2kml.cpp
index 4a63ed9..442297e 100644
--- a/tools/kml2kml/kml2kml.cpp
+++ b/tools/kml2kml/kml2kml.cpp
@@ -19,6 +19,7 @@
 #include <QApplication>
 #include <QFile>
 #include <QDebug>
+#include <QDataStream>
 
 using namespace std;
 using namespace Marble;
diff --git a/tools/stars/stars.cpp b/tools/stars/stars.cpp
index d3ba58c..ed810ce 100644
--- a/tools/stars/stars.cpp
+++ b/tools/stars/stars.cpp
@@ -16,6 +16,7 @@
 #include <QApplication>
 #include <QStringList>
 #include <QFileInfo>
+#include <QDataStream>
 
 #include <cmath>
 #include <iostream>
diff --git a/tools/svg2pnt/svgxmlhandler.cpp b/tools/svg2pnt/svgxmlhandler.cpp
index 1dfb3a9..8a59cd5 100644
--- a/tools/svg2pnt/svgxmlhandler.cpp
+++ b/tools/svg2pnt/svgxmlhandler.cpp
@@ -12,6 +12,7 @@
 
 #include "svgxmlhandler.h"
 #include <QDebug>
+#include <QDataStream>
 
 SVGXmlHandler::SVGXmlHandler(const QString& targetfile)
 {
